lib/src/element_type.dart

```diff
@@ -296,13 +300,12 @@ abstract class DefinedElementType extends ElementType {
   /// Return this type, instantiated to bounds if it isn't already.
   @override
   late final DartType instantiatedType = () {
-    if (_bound is InterfaceType &&
-        !(_bound as InterfaceType)
-            .typeArguments
-            .every((t) => t is InterfaceType)) {
+    final bound = _bound;
+    if (bound is InterfaceType &&
+        !bound.typeArguments.every((t) => t is InterfaceType)) {
       var typeSystem = library.element.typeSystem;
       return typeSystem.instantiateToBounds2(
-          classElement: _bound.element as ClassElement,
+          classElement: bound.element2 as ClassElement,
```

Comment:
Not every `InterfaceType.element2` is `ClassElement`.
Ah, but `instantiateToBounds2` requires `ClassElement`.
I will need to change it to `InterfaceElement`.

Reply:
Sounds good!